lib/scripting/post-process/expr.ts

```diff
@@ -17,138 +17,288 @@
+			index += node.children[loop].text.length + 1;
+		}
+	}
+	return estree;
 }
 
-export function mod(result: [Expression, null, Token, null, Expression]): Expression {
-	const leftExpr = result[0];
-	const rightExpr = result[4];
-	return estree.binaryExpression('%', leftExpr, rightExpr);
+function ppTypeExpression(node: ESIToken): any {
+	let estree = node.children[0].estree;
+	let index = node.children[0].text.length;
+	for (let loop = 1; loop < node.children.length; loop++) {
+		if (node.text.substr(index).startsWith(' Is ')) {
+			estree = binaryExpression('==', estree, node.children[loop].estree);
+			index += node.children[loop].text.length + 4;
+		}
+	}
+	return estree;
 }
 
-export function mult(result: [Expression, null, Token, null, Expression]): Expression {
-	const leftExpr = result[0];
-	const operator = result[2].text as BinaryOperator;
-	const rightExpr = result[4];
-	return estree.binaryExpression(operator, leftExpr, rightExpr);
+function ppUnaryExpression(node: ESIToken): any {
+	return unaryExpression(node.text.charAt(0) as UnaryOperator, node.children[0].estree);
 }
 
-export function unary(result: [Token, null, Expression]): Expression {
-	const operator = result[0].text as UnaryOperator;
-	const expr = result[2];
-	return estree.unaryExpression(operator, expr);
+function ppLogicalNotExpression(node: ESIToken): any {
+	return unaryExpression('!', node.children[0].estree);
 }
 
-export function concat(result: [Expression, null, Token, null, Expression]): Expression {
-	const leftExpr = result[0];
-	const operator = result[2].text as BinaryOperator;
-	const rightExpr = result[4];
-	return estree.binaryExpression('+', leftExpr, rightExpr);
+function ppParenthesizedExpression(node: ESIToken): any {
+	return node.children[1].estree;
 }
 
-export function is(result: [Expression, null, Token, null, Expression]): Expression {
-	const leftExpr = result[0];
-	const rightExpr = result[4];
-	return estree.binaryExpression('==', leftExpr, rightExpr);
+function ppMemberAccessExpression(node: ESIToken) {
+	return identifier('.' + node.children[1].estree.name);
 }
 
-export function isNot(result: [Expression, null, Token, null, Token, null, Expression]): Expression {
-	const leftExpr = result[0];
-	const rightExpr = result[6];
-	return estree.binaryExpression('!=', leftExpr, rightExpr);
+function ppSubExpression(node: ESIToken): any {
+	let id: any = null;
+	const argLists = [];
+	let expr = null;
+	let args;
+	for (const child of node.children) {
+		if (child.type === 'MemberAccessExpression' || child.type === 'SimpleNameExpression') {
+			id = child.estree;
+		} else if (child.type === 'OpenParenthesis') {
+			args = [];
+		} else if (child.type === 'ArgumentList') {
+			args = child.estree;
+		} else if (child.type === 'CloseParenthesis') {
+			argLists.push(args);
+		} else if (child.type === 'SubExpression') {
+			expr = child.estree;
+		}
+	}
+	let estree: any;
+	if (argLists.length > 0) {
```

Comment:
minor: `let estree: any = id;` then you can avoid the else loop

Reply:
Sounds good